pyiode/pyiode_model.pyx

```diff
@@ -6,45 +6,246 @@
+#                   inter_listname: str, 
+#                   post_listname: str, 
+#                   eqs_list = None)
+#  
+#   model_simulateparms => included int the parameters of model_simulate(), model_simulate_scc() and model_calc_scc()
+#  
+#   model_simulate_scc( sample_from: str, sample_to: str, 
+#                       pre_listname: str = "_PRE", 
+#                       inter_listname: str = "_INTER", 
+#                       post_listname: str = "_POST")
+                   
+#   model_exchange => included in the parameters of model_simulate()
+#   model_compile  => not yet implemented
+#  
+#  
+# model_simulate_maxit() -> int                 | Returns the maxit parameter of the last simulation   
+# model_simulate_eps() -> float                 | Returns the eps parameter of the last simulation
+# model_simulate_relax() -> float               | Returns the relax parameter of the last simulation
+# model_simulate_nb_passes() -> int             | Returns the nb_passes parameter of the last simulation
+# model_simulate_sort_algo() -> int             | Returns the sort_algo parameter of the last simulation
+# model_simulate_init_values() -> int           | Returns the init_values parameter of the last simulation 
+# model_simulate_cpu_scc() -> int               | Returns the elapsed time during the last SCC decomposition 
+# model_simulate_cpu_sort() -> int              | Returns the elapsed time during the last sort algorithm
+# model_simulate_cpu(period: str) -> int        | Returns the elapsed time during the last simulation of the given period
+# model_simulate_niter(period: str) -> int      | Returns the number of iterations needed to reach a solution during the last simulation of the given period
+# model_simulate_norm(period: str) -> float     | Returns the convergence threshold reached during the last simulation of the given period
+
+from pyiode_model cimport IodeModelSimulate, IodeModelCalcSCC, IodeModelSimulateSCC, IodeModelSimNorm, IodeModelSimNIter, IodeModelSimCpu, IodeModelCpuSort, IodeModelCpuSCC
+
+def model_simulate(sample_from: str, sample_to: str, 
+                    eqs_list=None, 
+                    endo_exo_list=None,
+                    eps: float = 0.0001, 
+                    relax: float = 1.0, 
+                    maxit: int = 100, 
+                    init_values: int = KV_INIT_TM1, 
+                    sort_algo: int = SORT_BOTH, 
+                    nb_passes: int = 5, 
+                    debug: int = 0, 
+                    newton_eps: float = 1e-6, 
+                    newton_maxit: int = 50, 
+                    newton_debug: int = 0):
     '''
     Simulate the model defined by eqs_list on the period [sample_from, sample_to].     
     '''
-    if type(eqs_list) == list:
+    if isinstance(eqs_list, list):
         eqs_list = ','.join(eqs_list)
     elif eqs_list is None:
         eqs_list = ""
 
-    if type(endo_exo_list) == list:
+    if isinstance(endo_exo_list, list):
```

Comment:
if isinstance(endo_exo_list, Iterable):

Reply:
Ok.

Reply:
See my comment https://github.com/plan-be/iode/pull/499#discussion_r1523727862